contrib/aws/aws-sdk-go-v2/aws/aws.go

```diff
@@ -101,6 +126,228 @@ func (mw *traceMiddleware) startTraceMiddleware(stack *middleware.Stack) error {
 	}), middleware.After)
 }
 
+func extractResourceNameFromParams(requestInput middleware.InitializeInput, awsService string) (string, string, error) {
+	var resourceNameKey, resourceNameValue string
+
+	switch awsService {
+	case "SQS":
+		resourceNameKey = tagQueueName
+		resourceNameValue = extractQueueName(requestInput)
+	case "S3":
+		resourceNameKey = tagBucketName
+		resourceNameValue = extractBucketName(requestInput)
+	case "SNS":
+		resourceNameKey = tagTopicName
+		resourceNameValue = extractTopicName(requestInput)
+	case "DynamoDB":
+		resourceNameKey = tagTableName
+		resourceNameValue = extractTableName(requestInput)
+	case "Kinesis":
+		resourceNameKey = tagStreamName
+		resourceNameValue = extractStreamName(requestInput)
+	case "EventBridge":
+		resourceNameKey = tagRuleName
+		resourceNameValue = extractRuleName(requestInput)
+	case "SFN":
+		resourceNameKey = tagStateMachineName
+		resourceNameValue = extractStateMachineName(requestInput)
+	default:
+		return "", "", fmt.Errorf("attemped to extract ResourceNameFromParams of an unsupported AWS service: %s", awsService)
+	}
+
+	return resourceNameKey, resourceNameValue, nil
+}
+
+func extractQueueName(requestInput middleware.InitializeInput) string {
+	var queueURL *string
```

Comment:
It's very strange to have string pointers in this way. Would prefer to just have this be a `string` type, then dereference the values below (assuming they will never be nil)